openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaAuthConfigGenerator.java

```diff
@@ -907,7 +981,11 @@ public String getAuthFieldTypeName() {
                     httpFieldTypeNames.add(AuthConfigTypes.BASIC.getValue());
                     break;
                 case CLIENT_CRED:
-                    httpFieldTypeNames.add(AuthConfigTypes.CLIENT_CREDENTIAL.getValue());
+                    if (clientCredGrantTokenUrl != null) {
+                        httpFieldTypeNames.add(AuthConfigTypes.CUSTOM_CLIENT_CREDENTIAL.getValue());
+                    } else {
+                        httpFieldTypeNames.add(AuthConfigTypes.CLIENT_CREDENTIAL.getValue());
```

Comment:
Shall we add a test case to cover this scenario ?

Reply:
Added